adapters/handlers/rest/dynamic_user/handlers_dynamic_users.go

```diff
@@ -170,6 +182,10 @@ func (h *dynUserHandler) deleteUser(params users.DeleteUserParams, principal *mo
 		return users.NewDeleteUserForbidden().WithPayload(cerrors.ErrPayloadFromSingleErr(err))
 	}
 
+	if h.staticUserExists(params.UserID) {
+		return users.NewDeleteUserBadRequest().WithPayload(cerrors.ErrPayloadFromSingleErr(fmt.Errorf("user %v is static user", params.UserID)))
```

Comment:
IMO, this should be a 422 rather than a 400 as a RESTful 400 means that the client request was incorrectly formatted whereas 422 is when the content cannot be processed processed. wdyt?